merge.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+
+def merge_sort(lst):
+    """
+    Sort a list of values using merge sort.
+    Return the sorted version of the list
+    """
+    if len(lst) <= 1:
+        return lst
+
+    middle = len(lst) // 2
+    left = lst[:middle]
+    right = lst[middle:]
+
+    sort_left = merge_sort(left)
+    sort_right = merge_sort(right)
+
+    return merge(sort_left, sort_right)
+
+
+def merge(left, right):
+    """
+    Helper function for merge sort. Merges two pre-sorted lists of values
+    """
+    sort = []
+    l_index = 0
+    r_index = 0
+    while l_index < len(left) and r_index < len(right):
+        if left[l_index] <= right[r_index]:
+            sort.append(left[l_index])
+            l_index += 1
+        else:
+            sort.append(right[r_index])
+            r_index += 1
+
+    if l_index < len(left):
+        sort.extend(left[l_index:])
```

Comment:
since slices of lists are safe (they won't return errors even if you slice beyond the end of the list), you don't really need the test here, you can just do `sort.extend(left[l_index:])` without worrying.